escher/rendering/model_renderer.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <glm/glm.hpp>
+
+#include "escher/base/macros.h"
+#include "escher/gl/unique_frame_buffer.h"
+#include "escher/scene/model.h"
+#include "escher/shaders/material/material_shader_factory.h"
+
+namespace escher {
+
+// Draws the contents of a model.
+class ModelRenderer {
+ public:
+  ModelRenderer();
+  ~ModelRenderer();
+
+  // Draws the content of the model to the specified frame buffer which
+  // must have color and depth buffers attached.
+  void DrawModel(const Model& model,
+                 const glm::mat4& matrix,
+                 const UniqueFrameBuffer& frame_buffer);
+
+ private:
+  class DrawContext {
```

Comment:
It stores private state within the ModelRenderer to retain cached information between renderings.  So I could make it is own class but they would end up being friends.